src/resolvers/contextual/workspace-resolver.js

```diff
@@ -19,19 +19,40 @@ export default class WorkspaceResolver extends BaseResolver {
 
   workspaceLayout: WorkspaceLayout;
 
-  resolve(): Promise<Manifest> {
+  resolve(downloadedManifest: ?Manifest): Promise<Manifest> {
     const workspace = this.workspaceLayout.getManifestByPattern(this.request.pattern);
     invariant(workspace, 'expected workspace');
     const {manifest, loc} = workspace;
+    if (manifest._remote && manifest._remote.registryRemote) {
+      return Promise.resolve(manifest); //already downloaded
+    }
     const registry = manifest._registry;
     invariant(registry, 'expected reference');
-
-    manifest._remote = {
+    let hash = '';
+    let registryRemote;
+    if (downloadedManifest && manifest.version === downloadedManifest.version) {
+      registryRemote = downloadedManifest._remote;
+      invariant(registryRemote, 'missing remote info');
+      hash = registryRemote.hash;
+      //override any local changes to manifest
```

Comment:
this is related to the comment [here](https://github.com/yarnpkg/rfcs/pull/92#issuecomment-375432881).